docs/src/tutorials/symbmc.md

```diff
@@ -0,0 +1,657 @@
+  y
+
+Init == x \in 1..3 /\ y = 2
+
+Next == x' = x + 1 /\ UNCHANGED y
+```
+we can deduce: 
+```tla
+R(0) = r(0) = S0 = {[x: 1, y: 2],[x: 2, y: 2],[x: 3, y: 2]}
+
+R(1) = {[x: 2, y: 2], [x: 3, y: 2], [x: 4, y: 2]}
+r(1) = {[x: 1, y: 2], [x: 2, y: 2], [x: 3, y: 2], [x: 4, y: 2]}
+
+R(2) = {[x: 3, y: 2], [x: 4, y: 2], [x: 5, y: 2]}
+r(2) = {[x: 1, y: 2], [x: 2, y: 2], [x: 3, y: 2], [x: 4, y: 2], [x: 5, y: 2]}
+```
+and so on. We can express this compactly as:
+```
+R(i) = [x: (i+1)..(i+3), y: {2}],
+r(i) = [x: 1..(i+3)    , y: {2}],
+R    = [x: N           , y: {2}]`
+```
+
+### Finite diameters
+We say that a transition system has a _finite diameter_, if there exists a `k \in N`, such that `R = r(k)`.
+
+If such an integer exists then the smallest integer `k`, for which this holds true, is the _diameter_ of the transition system.
+In other words, if the transition system `(S,S0,->)` has a finite diameter of `k`, any state that is reachable from a state in `S0`, is reachable in at most `k` transitions. The example above clearly does not have a finite diameter, since `R` is infinite, but `r(k)` is finite for each `k`.
+
+However, the spec
+```tla
+VARIABLE 
+  \* @type: Int;
+  x
+
+Init == x = 0 
+
+Next == x' = (x + 1) % 7
+```
+has a finite diameter (more specifically, a diameter of 6), because:
+  
+  1. `R = 0..6` (the set of remainders modulo 7), since those are the only values `x'`, which is defined as a `% 7` expression, can take. 
+  1. for any `k = 0,...,5`, it is the case that `r(k) = 0..k # R`, so the diameter is not in `{1,...,5}`
+  1. for any `k >= 6`, `r(k) = 0..6 = r(6) = R`
+
+### Invariants
+Much like `Init`, an invariant operator `Inv` defines a predicate. However, it is not, in general, the case that `Inv` defines a predicate over `S`. There are different cases we can consider, discussed in more detail [here](...). For the purposes of this document, we focus on _state invariants_, i.e. operators which use only unprimed variables and no temporal- or trace- operators. A state invariant operator `Inv` defines a predicate `I` over `S`.
+We say that the `I` is an _invariant_ in the transition system, if `R \subseteq I`, that is, for every reachable state `r \in R`, `I(r)` holds true. If `R \ I` is nonempty (i.e., there exists a state `r \in R`, such that `~I(r)`), we refer to elements of `R \ I` as _witnesses_ to invariant violation.
+
+### Goals of model checking
+The goal of model checking is to determine whether or not `R \ I` contains a witness. 
+The goal of bounded model checking is to determine, given a bound `k`, whether or not `r(k) \ I` contains a witness.
+
+In a transition system with a bounded diameter, one can use bounded model checking to solve the general model checking problem, since `R \ I` is equivalent to `r(k) \ I` for a sufficiently large `k`. In general, if the system does not have a bounded diameter, failing to find a witness in `r(k) \ I` cannot be used to reason about the absence of witnesses in `R \ I`! 
```

Comment:
I think the order of definitions in this text is not introductory. You could talk about model checking first. Then introduce bounded model checking, because that is what we can do with SMT. Then explain that bounded model checking is incomplete. Once you have this motivation, you could talk about the diameters and invariants.

I am actually not sure that you want to talk about diameters at all, perhaps, in the end of this text. It's cool when your system has a small diameter, but in practice it rarely does, so why do the readers need this?

Reply:
It's less about "small" vs "big" and more "finite" vs "infinite". The point is that you can't do much, if anything, with specs that define systems with infinite diameters, without BMC.